jina/drivers/search.py

```diff
@@ -75,8 +75,12 @@ def _apply_all(
             miss_idx = (
                 []
             )  #: missed hit results, some search may not end with results. especially in shards
-            for idx, retrieved_doc in enumerate(docs):
-                serialized_doc = self.exec_fn(retrieved_doc.id)
+            ids, _ = docs.extract_docs('id')
+            serialized_docs = self.exec_fn(ids)
```

Comment:
can u do the same here as in the `evaluate.py` driver?
```suggestion
            serialized_docs = self.exec_fn([d.id for d in docs])
```

Reply:
and remove the `extract_docs`?

Reply:
yes, good point.